instrumentation/executors/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/executors/ContextPropagatingRunnable.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.bootstrap.executors;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+
+public final class ContextPropagatingRunnable implements Runnable {
+
+  public static boolean shouldDecorateRunnable(Runnable task) {
+    // We wrap only lambdas' anonymous classes and if given object has not already been wrapped.
+    // Anonymous classes have '/' in class name which is not allowed in 'normal' classes.
```

Comment:
```suggestion
    // Anonymous classes have '/' in class name which is not allowed in 'normal' classes.
    // note: it is always safe to decorate lambdas since downstream code cannot be expecting a specific runnable implementation anyways
```